buster/retriever/deeplake.py

```diff
@@ -55,36 +107,27 @@ def get_topk_documents(
         else:
             raise ValueError("must provide either a query or an embedding")
 
-        if source is not None:
-            logger.info(f"Applying source {source} filter...")
-            filter = {"metadata": {"source": source}}
+        if self.use_tql:
+            assert self.exec_option == "compute_engine", "cant use tql without compute_engine"
+            tql_query = build_tql_query(query_embedding, sources=sources, top_k=top_k)
+            data = self.vector_store.search(query=tql_query)
```

Comment:
what is the advantage of tql if there is already a search function implemented ?

Reply:
not sure i understand the question - search is implemented using the TQL query directly - they support search with python functions but its extremely slow

Reply:
then is there any reason we need the python function version? We should always use the tql right?

Reply:
TQL is only supported for enterprise users so we stil lneed python for open-source free version